tools/oeedger8r/Emitter.ml

```diff
@@ -141,43 +106,48 @@ let open_file (filename : string) (dir : string) =
   fprintf os " */\n" ;
   os
 
-(** [oe_mk_struct_decl] constructs the string of a [struct] definition. *)
-let oe_mk_struct_decl (fs : string) (name : string) =
-  String.concat "\n"
-    [ sprintf "typedef struct _%s" name
-    ; "{"
-    ; "    oe_result_t _result;"
-    ; sprintf "%s} %s;" fs name ]
-
 (** [oe_gen_marshal_struct_impl] generates a marshalling [struct]
     definition. *)
-let oe_gen_marshal_struct_impl (fd : func_decl) (errno : string)
-    (isecall : bool) =
-  (* TODO: Clean this up. *)
-  let member_list_str =
-    errno
-    ^
-    let new_param_list = List.map conv_array_to_ptr fd.plist in
-    List.fold_left
-      (fun acc (pt, declr) -> acc ^ mk_ms_member_decl pt declr isecall)
-      "" new_param_list
+let oe_gen_marshal_struct =
+  let gen_member_decl (ptype : parameter_type) (decl : declarator) =
+    let tystr = get_tystr (get_param_atype ptype) in
+    let tystr =
+      if is_foreign_array ptype then
+        sprintf "/* foreign array of type %s */ void*" tystr
+      else tystr
+    in
+    (* TODO: We shouldn't have to emit a strlen variable for ocalls,
```

Comment:
It is simpler to have the marshaling strategy be consistent for ecalls and ocalls.
Is the concern about the length variables for ocalls in regards to space?

Reply:
Sorry, my comment wasn't very clear. Intel states [here](https://github.com/microsoft/openenclave/blob/6b5f1612ab124d05cba596f48b80558c4da36864/tools/oeedger8r/intel/CodeGen.ml#L324) a reason not to do this for ocalls, but I (briefly) tested removing it for us, and it didn't work, it's just a TODO to revisit.

Reply:
I don't think that reason is valid. Strings can be **in-out** parameters for ocalls, which is almost like an in parameter for ecalls.